modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -216,12 +217,37 @@ export default class MVTLayer extends TileLayer {
     return renderedFeatures;
   }
 
+  getVisibleTiles(tileCoords = 'local') {
+    const {selectedTiles} = this.state.tileset;
+    const {viewport} = this.context;
+
+    if (conversionToWGS84isRequired(tileCoords, viewport.resolution)) {
+      this._convertTilesToWGS84(selectedTiles, viewport);
+    }
+
+    return selectedTiles;
+  }
+
+  _convertTilesToWGS84(tiles, viewport) {
+    const {tileIdsCache} = this.state;
+
+    tiles.forEach(tile => {
+      const tileId = `${tile.x},${tile.y},${tile.z}`;
+
+      if (!tileIdsCache.has(tileId)) {
+        tileIdsCache.add(tileId);
+        tile.transformToWorld = () => updateTileWithWGS84Coords(tile, tile.content, viewport);
```

Comment:
This should be added when the tile is created, not when `getVisibleTiles` is called. Tiles can be accessed in other callbacks, e.g. `onTileLoad`, `renderSubLayers`

Reply:
Should we move the calculation out of `MVTLayer`?

Reply:
Better to move it inside `getTileData` `MVTLayer` method?